services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -2413,50 +2413,89 @@ public void updateStore(String clusterName, String storeName, UpdateStoreQueryPa
             "Cannot convert store to batch-only and enable incremental push together.",
             ErrorType.BAD_REQUEST);
       }
-      // Update active-active replication config.
+
+      // Update active-active replication config based on the input if present, otherwise use the current value.
       setStore.activeActiveReplicationEnabled = activeActiveReplicationEnabled
           .map(addToUpdatedConfigList(updatedConfigsList, ACTIVE_ACTIVE_REPLICATION_ENABLED))
           .orElseGet(currStore::isActiveActiveReplicationEnabled);
-      // Enable active-active replication automatically when batch user store being converted to hybrid store and
-      // active-active replication is enabled for all hybrid store via the cluster config
-      if (storeBeingConvertedToHybrid && !setStore.activeActiveReplicationEnabled && !currStore.isSystemStore()
+      // Update incremental push config.
+      setStore.incrementalPushEnabled =
+          incrementalPushEnabled.map(addToUpdatedConfigList(updatedConfigsList, INCREMENTAL_PUSH_ENABLED))
```

Comment:
I am wondering, if you move it here, would it be the case that it will enable inc push if admin tool ask it to do so, regardless of A/A or hybrid config it has? Should we allow this?

Reply:
True. It will also allow us to specifically disable IncPush and keep AA.

Reply:
This change was removed.